Security improvement and other UI fixes

nessv opened the pull request:
# Changes <!-- [REQIURED] -->
<!-- Fill in a bulleted list with changes made in this pull request -->
- Fixed misspelling strings
- Added security to avoid unauthorize user to login
- Changed the order that indicators appear in the life map.
- Improved handling of the prioritize indicators. To asure both Android and Web apps will be able to handle them correctly.

# Issues <!-- [IF APPLICABLE] -->
<!-- Fill in a bulleted list with issues that have been introduced or fixed -->
- Any user can login
- Some priorities weren't display correctly.

# Tests <!-- [REQUIRED] -->
<!-- Check off the following tests (using [X]) that you performed to ensure that your changes work -->
This code has been tested in the following ways:
- Tasks:
  - [X] Logged in to the application
  - [X] Fill out a snapshot for a new family
  - [X] Fill out a snapshot for an existing family
  - [X] Fill out priorities for a snapshot
  - [X] View a family and it's priorities
- API Level:
  - [ ] API Level 19
  - [ ] API Level 22
  - [X] API Level 25
- Screen Size:
  - [X] 7" screen size
  - [ ] 8" screen size
  - [ ] 10" screen size

nessv committed:
Addresses the hiccup when transitioning views

There was a hiccup when transitioning from the main view to some of the
families, because of frames been droped.

nessv committed:
Beta release build 13

nessv committed:
Fixes a couple of minor UI bugs

This includes:
- Misspelling in "Tu respuestas".
- Indicators in the lifemap are now presented in the same order that they appear in the survey.

nessv committed:
Prevents unauthorized users from login

Currently the app is target only to users with the role of "user_survey".
So a validation was added in order to prevent users with others roles to login.

nessv committed:
Beta release V14

Login security flaw fixed

nessv committed:
Fixes priorites bad sync

By changing the remote ID for the snapshot_priorities

nessv committed:
Changes the way priorities are handle

Now there is new parameter in the table where the prioritize indicators are stored,
the "json_key". Wich corresponds to the unique identifier of each question in the
json survey definition

alefq commented on app/src/main/java/org/fundacionparaguaya/adviserplatform/data/remote/AuthenticationManager.java:
```diff
@@ -207,6 +220,19 @@ private AuthenticationStatus updateLogin(@Nullable User user, retrofit2.Response
+         * That why we should iterate over all the values that
+         * @getAuthority() retrieves*/
+        for (UserRole role: roles) {
+            if (role.getAuthority().equals(AppConstants.USER_ROLE) ) {
```
Always use constant as left operator .

alefq reviewed (changes requested):
Please change PR title. Put as title the most important or two of the most relevant changes, then add "...and other minor fixes"

nessv commented:
@alefq ya actualicé mi PR.

nessv committed:
Replaces the priorities label

The displayed label for the priorities previously corresponded to the title
of the survey question. This was replace for the question's description.
In the same way is the description the one it gets stored in the database

nessv committed:
Refactorizes PR to Codacy standards
